src/core/geometry/qgsgeometryutils.h

```diff
@@ -244,7 +244,18 @@ class CORE_EXPORT QgsGeometryUtils
      * If the return value is 0, then the test was unsuccessful (e.g. due to testing a point exactly
      * on the line, or exactly in line with the segment) and the result is undefined.
      */
-    static int leftOfLine( double x, double y, double x1, double y1, double x2, double y2 );
+    static int leftOfLine( const double x, const double y, const double x1, const double y1, const double x2, const double y2 );
+
+    /**
+     * Returns a value < 0 if the point \a point is left of the line from \a p1 -> \a p2.
+     * A positive return value indicates the point is to the right of the line.
+     *
+     * If the return value is 0, then the test was unsuccessful (e.g. due to testing a point exactly
+     * on the line, or exactly in line with the segment) and the result is undefined.
+     *
+     * \since QGIS 3.6
+     */
+    static int leftOfLine( const QgsPoint &point, const QgsPoint &p1, const QgsPoint &p2 );
```

Comment:
I am not a huge fan of a method having `point`, `p1`and `p2` as arguments, but have no great idea.
Also, it is coherent with the other one, so do as you prefer!

Reply:
It is true that it is not explicit, but it is widely used in this class with other forms (cp, pt, etc.). Personally, this does not bother me in cases like this. I leave it like that.